### PR TITLE
Update gitops template

### PR DESCRIPTION
**Pull request by 6za**
Adding support to reported issues:
- https://github.com/kubefirst/kubefirst/issues/335
- https://github.com/kubefirst/kubefirst/issues/332
- https://github.com/kubefirst/kubefirst/issues/331

### CODE REVIEW

File: components/argo/argo.yaml [hunk 1]
```diff
@@ -554,8 +554,8 @@ subjects:
   name: argo-server
   namespace: argo
 - kind: ServiceAccount
-  name: default
-  namespace: gitlab-runner
+  name: github-runner
+  namespace: github-runner
```
Comment (6za) [inline components/argo/argo.yaml:558]: @pagottoo This is only for github version

File: components/argo/cwfts/cwft-git.yaml [hunk 2]
```diff
@@ -4,6 +4,51 @@ metadata:
...
+      - name: gitops-source
+        path: /src/gitops
+        git:
+          repo: "ssh://git@github.com/org-demo-6za/gitops.git"
+          revision: "main"
+          insecureIgnoreHostKey: true
+          sshPrivateKeySecret:
+            name: ci-secrets
+            key: SSH_PRIVATE_KEY    
+
```
Comment (6za) [inline components/argo/cwfts/cwft-git.yaml:40]: @pagottoo Missing on terraform

File: components/github-runner/github-runner.yaml [hunk 3]
```diff
@@ -1,6 +1,71 @@
...
+- apiGroups: [""]
+  resources: ["pods/log"]
+  verbs: ["get", "list", "watch",]
+- apiGroups: ["batch"]
+  resources: ["jobs"]
+  verbs: ["get", "list", "create", "delete"]
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "list", "create", "delete"]
+
```
Comment (6za) [inline components/github-runner/github-runner.yaml:41]: I think we don't need this anymore.. just the github-runner SA is enough.

File: components/github-runner/github-runner.yaml [hunk 4]
```diff
@@ -1,6 +1,71 @@
...
+metadata:
+  name: actions-runner
+subjects:
+- kind: ServiceAccount
+  name: actions-runner
+  namespace: github-runner
+roleRef:
+  kind: ClusterRole
+  name: actions-runner
+  apiGroup: rbac.authorization.k8s.io
```
Comment (6za) [inline components/github-runner/github-runner.yaml:55]: I think this is not need too.

**Review by pagottoo — APPROVED** [review]

File: components/argo/cwfts/cwft-git.yaml (hunk 2 above)
Comment (6za) [reply components/argo/cwfts/cwft-git.yaml:40]: https://github.com/kubefirst/kubefirst/issues/337

**Pull request merged**